torch/csrc/autograd/python_function.cpp

```diff
@@ -53,27 +53,6 @@ PyObject* THPGradientEdgeClass = nullptr;
     throw python_error();                  \
   }
 
-// Anonymous namespace for helpful functions used in this file
-namespace {
-
-// Throw a python_error with the PyErr state persisted, so that we
-// don't lose the error state if the GIL is released when we don't
-// have a PyThreadState created beforehand, this is made so that
-// even for pure C++ thread without a pre-created PyThreadState could
-// also capture the correct error message.
-// TODO: This is a temporary approach to allow C++ thread to correctly
-// capture Python Error in autograd, remove this when c10 thread pool
-// allow to do one time initialization.
-// see discussion in https://github.com/pytorch/pytorch/pull/34845
-// Follow up issue: https://github.com/pytorch/pytorch/issues/35006
-void throw_python_error() {
-  python_error err;
-  err.persist();
-  throw std::move(err);
-}
-
-} // namespace
-
```

Comment:
Not sure I can remove this for the dist engine actually, still need to investigate why.

Reply:
Not doing this anymore in this PR, left a note.